Generoi v2

oxyc opened the pull request.

oxyc committed:
add genero components

kalenjohnson committed:
Unused use keyword

kalenjohnson committed:
Merge branch 'master' of github.com:roots/sage

kalenjohnson committed:
First pass at Blade templating

QWp6t committed:
Resolve style issues

QWp6t committed:
Remove wrapper and template classes

QWp6t committed:
Require PHP >= 5.6.4

QWp6t committed:
Use array_map() for blade hierarchy loop

oxyc commented on templates/pages/index.twig:
```diff
@@ -0,0 +1,11 @@
+    {% block term %}{% endblock %}
+
+    {% block listing %}
+      {{parent()}}
```
This is nice

toffebjorkskog replied:
is this like super()

oxyc replied:
Jepp. Så kan man overridea layouten men ta int subdelar.

oxyc commented on templates/layout/hero.twig:
```diff
@@ -0,0 +1,58 @@
+{% set object = post ? post : (term ? term : false) %}
+{% set slides = object.get_field('hero_slide') %}
+{% set force_crop = object.get_field('hero_crop') or (slides|length == 0) %}
```
There you go @toffebjorkskog

toffebjorkskog replied:
Tackar!

oxyc commented on templates/layout/header.twig:
```diff
@@ -0,0 +1,79 @@
+    data-toggler
+    data-animate="slide-in-down slide-out-up fast"
+    style="display: none;"
+  >
```
off-canvas fick en revamp, men eftersom de e lite svårt att få de fint out of the box så iddes jag inte implementera det. Så många sidor som inte använder det att wrappers stör mer än hjälper.

retlehs committed:
Remove PHP 5.5 testing from Travis

oxyc commented on templates/layout/footer.twig:
```diff
@@ -0,0 +1,14 @@
+<footer id="footer" class="l-footer" role="contentinfo">
+  <div class="row align-center">
+    <div class="column shrink">
+      <p><small>{{options.footer_copyright}}</small></p>
```
Våra Customizer juttun e borttagna. Enklare att konfigurera nya fält med ACF.

oxyc commented on src/lib/Genero/PostTypeConnection/Menu.php:
```diff
@@ -0,0 +1,25 @@
+        if (is_null($post)) {
+            $post = get_post();
+        }
+        return \Genero\Sage\PostTypeConnection::isParent($pid, $post->post_type);
```
Dehär gör så att man kan skapa post type connections och menyn ser att det är child pages. Eg att alla posts tillhör blog sidan. Jag använder också samma för password inhertiance. Eg kan man ha en password protected page, och sen har den e lista med poster av en typ. Alla posterna kräver också password protection isf.

oxyc commented on src/custom/widgets.php:
```diff
@@ -0,0 +1,25 @@
+        'WP_Widget_Links',
+        'WP_Widget_Meta',
+        'WP_Widget_Search',
+        'WP_Widget_Text',
```
@petterw03 du brukar väl använda dessa ibland? Kan löna sig att testa ACF Widgets eftersom du då kan skapa va för fält som helst till widgeten. Du kan också fixa twig template åt dom, de finns ett exempel på de i koden också. Om du vill att jag lämnar kvar någon widget så kan vi nog bra göra det.

petterw03 replied:
Text Widget e den enda av de default jag brukar använda, annars brukar jag göra egna.

Men med sidor med WPML (nästan alla) är det ju Multilingual Text Widget som oftast gäller.

oxyc replied:
Ok, jo varenda WP sida jag byggt ha varit multilingual. Är det ok om jag lämnar de såhär? Så kan man antingen:
1. Skapa en egen Text widget med ACF Widgets (och kunna använda namnet)
2. Ändra denna hook

oxyc commented on src/custom/timber.php:
```diff
@@ -0,0 +1,90 @@
+    // {{ post.phone|format_number }}
+    $twig->addFilter('format_phone', new Twig_SimpleFilter('format_phone', function ($number) {
+        return App\Utils\format_phone($number);
+    }));
```
Flesta filter och funktioner e flyttade till https://github.com/generoi/wp-timber-extended

oxyc commented on src/custom/setup.php:
```diff
@@ -0,0 +1,108 @@
+add_action('widgets_init', function () {
+    $config = [
+        'before_widget' => '',
+        'after_widget'  => '',
```
@petterw03 angående core widgets, se den här också.

oxyc commented on src/custom/setup.php:
```diff
@@ -0,0 +1,108 @@
+     */
+    add_theme_support('timber-extended-templates', [
+        // Use double dashes as the template variation separator.
+        'bem_templates',
```
Både standard templates `archive-category` och BEMified `archive--category` fungerar.

toffebjorkskog replied:
Nice

oxyc commented on package.json:
```diff
@@ -70,8 +70,17 @@
+    "fastdom": "^1.0.5",
     "font-awesome": "^4.7.0",
-    "jquery": "1.12.4 - 3"
+    "foundation-sites": "^6.3.0-rc1",
```
behöver ännu https://github.com/zurb/foundation-sites/pull/9449

oxyc commented on assets/styles/utils/_typography.scss:
```diff
@@ -0,0 +1,99 @@
+/**
+ * Based on http://typecast.com/blog/a-more-modern-scale-for-web-typography.
+ * If you choose to use this, move it to _variables().
+ */
```
Yuki använde sig av detta på teritalot nya sida. Kanske vi skulle försöka få fler designers att göra det? En ganska bra och väl uttänkt bas.

toffebjorkskog replied:
Shared it with Jeppis designers

oxyc commented on assets/styles/utils/_spacings.scss:
```diff
@@ -0,0 +1,81 @@
+%u-no-spacing {
+  @extend %u-no-spacing--vertical;
+  @extend %u-no-spacing--horizontal;
+}
```
Jag har tagit bort alla `.margin-top` etc klasser. Tanken är att spacing är komponent attribut, och inte styling variationer. Tror att det kan göra de lite lättare att förstå stylingen om de int finns en massa css klasser som ändrar utseende.

oxyc commented on assets/styles/utils/_spacings.scss:
```diff
@@ -0,0 +1,81 @@
+    small: rem-calc(40) rem-calc(10),
+    medium: rem-calc(60) rem-calc(10),
+    large: rem-calc(75) rem-calc(60),
+  ),
```
Tanken är att man ska försöka hålla sig till existerande mått istället för att sätta massa random paddings överallt. Här kan t.ex. `post` vara en spacing definition, eller `blockquote`. Men ofta kan säkert gutter återanvändas.

oxyc commented on assets/styles/components/foundation/_orbit.scss:
```diff
@@ -0,0 +1,47 @@
+    }
+  }
+  &-previous:before { content: $fa-var-chevron-left; }
+  &-next:before { content: $fa-var-chevron-right; }
```
Jag orkar inte fixa custom font builds mer. Buggar för mycket. Nu finns hela fontawesome med och man kan använda det såhär.

oxyc commented on assets/styles/components/genero/_grid.scss:
```diff
@@ -0,0 +1,28 @@
+        margin-top: 0;
+      }
+    }
+  }
```
<3 sass

oxyc commented on assets/build/webpack.config.watch.js:
```diff
@@ -16,6 +16,11 @@ module.exports = {
       watch: config.watch,
+      rsync: config.rsync,
+      browserSyncOptions: {
+        reloadDebounce: 800,
```
Om det visar sig att browsern refreshas innan en php/twig fil rsyncats klart (tiden visas i output) så kan man höja det här. För mig ligger det alltid under 800ms dock.

oxyc commented on .eslintrc:
```diff
@@ -2,7 +2,10 @@
+    "wp": true,
+    "picturefill": true,
+    "Foundation": true,
+    "fastdom": true
```
Lägger ni till någo js libs, måst ni definiera deras globals här så att int eslint ska klaga.

oxyc commented on assets/scripts/main.js:
```diff
@@ -1,6 +1,8 @@
 // import external dependencies
 import 'jquery';
-import 'bootstrap/dist/js/bootstrap';
+import 'picturefill';
```
```sh
npm install picturefill --save
```
och sen e de bara att kör import (samt definiera dess global i `.eslintrc`)

QWp6t committed:
Refactor Blade implementation. Use Container

retlehs committed:
Update README.md

retlehs committed:
Uncomment sidebar, fix include

retlehs committed:
:eyes:

kalenjohnson committed:
Fix sidebar echoing a 1

kalenjohnson committed:
Fixing double echo's, template cleanup

retlehs committed:
Merge pull request #1765 from roots/blade

Blade

toffebjorkskog reviewed (commented):
Ser bra ut

QWp6t committed:
Ignore yarn-error.log file 👀

aaemnnosttv committed:
tighten version constraint for illuminate/view (#1767)

QWp6t committed:
Use hyphen in view paths as delimiter for fallback views

QWp6t committed:
Merge branch 'master' of github.com:roots/sage

QWp6t committed:
Fallback delimiter should be const

QWp6t committed:
Use @php() instead of @php [...] @endphp

QWp6t committed:
Normalize $templates[] in {$type}_template_hierarchy

QWp6t committed:
Add `sage/template_data` filter so developers can pass data within template_include

QWp6t committed:
Array manipulation in FileViewFinder::getPossibleViewFiles() should be a separate method

QWp6t committed:
Update .travis.yml for Sage 9

QWp6t committed:
Update composer.lock

retlehs committed:
npm -> yarn

retlehs committed:
npm -> yarn

davekiss committed:
front_page => frontpage

Correct the value based on what WordPress passes as filter values

https://github.com/WordPress/WordPress/blob/4.7-branch/wp-includes/template.php#L29-L41

retlehs committed:
Merge pull request #1772 from davekiss/patch-1

front_page => frontpage

oxyc commented:
Behöver säkert bygga en sida till innan det här känns ok. Kan i något skede ta dev-möte, och så kan vi se om det är något ni tycker inte ska vara med här (eg sections kanske inte alla gillar).

toffebjorkskog commented:
Det låter bra!

QWp6t committed:
Refactor Blade implementation (again), closes #1769 (#1777)

* Squash bugs, reorganize, etc.
* Use `get_body_class()` to apply filters on template data
* Use `PHP_INT_MAX` as priority for `template_include` filter

QWp6t committed:
Pass `$template` to `sage/template/{$class}/data` filter, closes #1771

QWp6t committed:
Do not use `after_switch_theme` hook, closes #1768

QWp6t committed:
Remove `templates/` when normalizing templates in hierarchy filters

QWp6t committed:
Use Config object to store config options

QWp6t committed:
Prefer passing class name when binding new shared instances

retlehs committed:
Enable selective refresh for widgets in customizer (#1778)

QWp6t committed:
Use `get_theme_file_path()`

QWp6t committed:
Merge branch 'master' of github.com:roots/sage

QWp6t committed:
Throw errors if minimum requirements aren't met (#1782)

QWp6t committed:
Merge branch 'master' into config-class

retlehs committed:
Merge pull request #1781 from roots/config-class

Use Config object to store config options

retlehs committed:
Fix conflict with WP core

QWp6t committed:
Use browsersync-webpack-plugin

retlehs committed:
Merge pull request #1787 from roots/qwp6t/browsersync-webpack-plugin

Use browsersync-webpack-plugin

retlehs committed:
Update to Bootstrap 4 Alpha 6

retlehs committed:
Merge pull request #1792 from roots/bs4-alpha-6

Update to Bootstrap 4 Alpha 6

retlehs committed:
Close #1791 - Only disable customizer if theme isn't activated

retlehs committed:
Merge pull request #1793 from roots/customizer-fix

Only disable customizer if theme isn't activated

retlehs committed:
👀

QWp6t committed:
Update dependencies

QWp6t committed:
Update webpack 2.2.0-rc.3

QWp6t committed:
Update airbnb 14.0.0

QWp6t committed:
`publicPath` is `../` for styles, and optionally uses envar for everything else

QWp6t committed:
images should be recompiled on compiler `done` event

QWp6t committed:
Follow standard naming conventions for classes/non-classes in assets/scripts

QWp6t committed:
Give devs more options for dynamically setting `publicPath`

QWp6t committed:
Bump version 9.0.0-beta.1

QWp6t committed:
update eslint 3.13.1

swalkinshaw committed:
Use stock eslint

QWp6t committed:
Add shared `$post` variable to Blade

retlehs committed:
alpha -> beta

retlehs committed:
Update CHANGELOG.md

retlehs committed:
Merge pull request #1795 from roots/beta1

Beta1

oxyc committed:
Allow autoprefixer's browser versions to be configured

oxyc committed:
Respect autoprefixer browsers when stripping vendor prefixes with cssnano

psorensen committed:
rename clean task

#1797

psorensen committed:
Update README to include new task name

#1797

psorensen committed:
Update travis

#1797

swalkinshaw committed:
Merge pull request #1796 from roots/use-stock-eslint

Use stock eslint

retlehs committed:
Fix for publicPath

retlehs committed:
Merge pull request #1800 from psorensen/issue/1797

Update clean task name

retlehs committed:
Merge pull request #1798 from oxyc/patch-1

Allow browser versions to be configured in config.json

retlehs committed:
Use NoEmitOnErrorsPlugin, thanks @eshimischi

retlehs committed:
Update CHANGELOG, bump version [ci skip]

retlehs committed:
Revert to yarn.lock from master, previous Webpack RC

retlehs committed:
Remove outdated browser warning

retlehs committed:
Remove grid defaults

retlehs committed:
Merge pull request #1808 from roots/remove-grid-defaults

Remove grid defaults

retlehs committed:
Remove Font Awesome

retlehs committed:
Update CHANGELOG [ci skip]

retlehs committed:
Merge pull request #1809 from roots/remove-font-awesome

Remove Font Awesome

retlehs committed:
Add option to remove Bootstrap when creating project (#1810)

* Add option to remove Bootstrap when creating project
* Remove phpmd, per @QWp6t

retlehs committed:
Missed in #1809

retlehs committed:
Add option to change theme file headers

retlehs committed:
Update README, CHANGELOG [ci skip]

retlehs committed:
Merge pull request #1811 from roots/option-to-change-theme-file-headers

Add option to change theme file headers

retlehs committed:
Add option to add Font Awesome

retlehs committed:
Merge pull request #1812 from roots/option-to-add-font-awesome

Add option to add Font Awesome

retlehs committed:
Add option to select CSS framework, add Foundation as an option (#1813)

* Add option to select CSS framework, add Foundation as an option

QWp6t committed:
Fix browsersync (#1815)

* Update dependencies
* Remove monkey-hot-loader
* Use webpack-merge instead of util/mergeWithConcat()
* Fix: copyglobs plugin `after-emit` is bound twice
* NoErrorsPlugin() is deprecated in favor of NoEmitOnErrorsPlugin()
* webpack-dev-middleware should use same publicPath as compiler
* webpack-hot-middleware/client should be prepended to entries
* Browser should refresh when HMR fails
* Bootstrap package.json has correct main property
Sometime between alpha 2 and 3, package.json was pointing to nonexistent
file, so we referenced file manually in our repo. Underlying issue was
fixed by alpha 4, so we no longer have to specify full path to file.

retlehs committed:
Fix Bootstrap JS removal/replacement

retlehs committed:
Update CHANGELOG [ci skip]

retlehs committed:
Merge pull request #1806 from roots/beta2

Beta 2

QWp6t committed:
Add support for HTML injection (#1817)

Commit:
Write modified package.json with Font Awesome.

Fixes #1818.

retlehs committed:
Merge pull request #1819 from greatislander/fix-font-awesome

Write modified package.json with Font Awesome (fix #1818)

retlehs committed:
Update README [ci skip]

retlehs committed:
Fix BS4 image caption padding

retlehs committed:
Add option to configure Browsersync

tobeycodes committed:
Update functions.php (#1821)

cedarstay committed:
Update screenshot dimensions

retlehs committed:
Merge pull request #1827 from roots/update-screenshot-dimensions

Update screenshot dimensions

retlehs committed:
Not specific to BS

retlehs committed:
Merge pull request #1822 from roots/browsersync-install-options

Add option to configure build settings

retlehs committed:
This shouldn't have had the slash [ci skip]

QWp6t committed:
Comment out instructions in ISSUE_TEMPLATE.md

QWp6t committed:
Use copy-globs-webpack-plugin (#1824)

kaisermann committed:
Fixed typo on readme

swalkinshaw committed:
Merge pull request #1836 from kaisermann/readme-typo

Fix typo in README

LeoColomb committed:
do not call composer package binaries with path (#1846)

Fixes #1842
See composer note at https://getcomposer.org/doc/articles/scripts.md#writing-custom-commands

QWp6t committed:
Resolve variables passed to @asset() directive

Fixes #1845

drawcard committed:
Add requirement note for Wordpress 4.7

Referencing: https://discourse.roots.io/t/undefined-function-app-get-theme-file-path/8455

slackday committed:
Add environment variable check for DEVURL

Commit:
Upgrade to Illuminate 5.4 packages #1841 (#1850)

retlehs committed:
Merge pull request #1847 from drawcard/patch-2

Add requirement note for Wordpress 4.7

retlehs committed:
Merge pull request #1849 from slackday/PR/dynamic-devurl

Check for dynamic env DEVURL before config

rfair404 committed:
add version check for WordPress version less than 4.7.0

swalkinshaw committed:
Merge pull request #1852 from rfair404/rfair/wp-47-version-check

Add version check for WordPress version less than 4.7.0

Commit:
Fix wp_link_pages displayed twice

oxyc committed:
disable sage filters

oxyc committed:
remove section component

QWp6t committed:
Fix injection issues

QWp6t committed:
Update dependencies

retlehs committed:
Merge pull request #1854 from plaurent75/fix-wp-link-pages

wp_link_pages displayed twice

QWp6t committed:
Update browsersync-webpack-plugin

QWp6t committed:
Enable HTML injection over SSL

oxyc committed:
fix rsync-watcher

oxyc committed:
various fixes

oxyc committed:
add a spinner
